packages/workbox-precaching/src/PrecacheController.ts

```diff
@@ -356,6 +426,36 @@ class PrecacheController {
     return undefined;
   }
 
+  /**
+   * Creates a [`matchCallback`]{@link module:workbox-precaching~matchCallback}
+   * based on the passed configuration options) that with will identify
+   * requests in the precache and return a respective `params` object
+   * containing the `cacheKey` of the precached asset.
+   *
+   * This `cacheKey` can be used by a
+   * [`handlerCallback`]{@link module:workbox-precaching~handlerCallback}
+   * to get the precached asset from the cache.
+   *
+   * @param {Object} [options] See
+   * [addRoute() options]{@link module:workbox-precaching.PrecacheController#addRoute}.
+   */
+  createMatchCallback(options?: FetchListenerOptions) {
+    return ({request}: RouteMatchCallbackOptions) => {
+      const urlsToCacheKeys = this.getURLsToCacheKeys();
+      for (const possibleURL of generateURLVariations(request.url, options)) {
+        const cacheKey = urlsToCacheKeys.get(possibleURL);
+        if (cacheKey) {
+          return {cacheKey};
```

Comment:
cc: @jeffposnick not sure how you feel about passing the `cacheKey` from the match callback to the handler callback. Originally I assume this logic would live in a plugin, but since it needs to be run in the match callback, I figure it's better to not have to run it twice. Let me know what you think.

Reply:
That sounds like it's in keeping with the intended purpose of `matchCallback` return values, so I'm 👍